ts/Extensions/Data.ts

```diff
@@ -2450,6 +2457,22 @@ export default Data;
  * @apioption data.columns
  */
 
+/**
+ * An array option that specifies the data type for each column in the series
+ * loaded within the data module.
+ *
+ * Possible values: `"string"`, `"number"`, `"date"`.
```

Comment:
To help our future Editor (and other GUI generators), we should add:

* A stricter [type](https://github.com/highcharts/highcharts/blob/e6cec31e422a680d9e2a6cc22b10f321d0c90d98/ts/Extensions/Data.ts#L87): `columnTypes?: Array<'string'|'number'|'date'>;
* A `@validvalue` doclet tag, even though this one will soon be redundant as we create the TS-driven API.